compute/instances/preventing-accidental-vm-deletion/preventing_accidental_vm_deletion_test.go

```diff
@@ -0,0 +1,105 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package snippets
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"math/rand"
+	"strings"
+	"testing"
+	"time"
+
+	compute "cloud.google.com/go/compute/apiv1"
+	"github.com/GoogleCloudPlatform/golang-samples/internal/testutil"
+	computepb "google.golang.org/genproto/googleapis/cloud/compute/v1"
+)
+
+func TestPreventingAccidentalVMDeletionSnippets(t *testing.T) {
+	ctx := context.Background()
+	instancesClient, err := compute.NewInstancesRESTClient(ctx)
+	if err != nil {
+		t.Fatalf("NewInstancesRESTClient: %v", err)
+	}
+	defer instancesClient.Close()
+
+	var seededRand *rand.Rand = rand.New(
+		rand.NewSource(time.Now().UnixNano()))
+	tc := testutil.SystemTest(t)
+	zone := "europe-central2-b"
+	instanceName := "test-instance-" + fmt.Sprint(seededRand.Int())
+
+	buf := &bytes.Buffer{}
+
+	if err := createInstance(buf, tc.ProjectID, zone, instanceName, true); err != nil {
+		t.Fatalf("createInstance got err: %v", err)
+	}
+
+	expectedResult := "Instance create"
+	if got := buf.String(); !strings.Contains(got, expectedResult) {
+		t.Errorf("createInstance got %q, want %q", got, expectedResult)
+	}
+
+	buf.Reset()
+
+	if err := getDeleteProtection(buf, tc.ProjectID, zone, instanceName); err != nil {
+		t.Fatalf("getDeleteProtection got err: %v", err)
+	}
+
+	expectedResult = fmt.Sprintf("Instance %s has DeleteProtection value: %v", instanceName, true)
+	if got := buf.String(); !strings.Contains(got, expectedResult) {
+		t.Errorf("getDeleteProtection got %q, want %q", got, expectedResult)
+	}
+
+	buf.Reset()
+
+	if err := setDeleteProtection(buf, tc.ProjectID, zone, instanceName, false); err != nil {
+		t.Fatalf("setDeleteProtection got err: %v", err)
+	}
+
+	expectedResult = "Instance updated"
+	if got := buf.String(); !strings.Contains(got, expectedResult) {
+		t.Errorf("setDeleteProtection got %q, want %q", got, expectedResult)
+	}
+
+	buf.Reset()
+
+	if err := getDeleteProtection(buf, tc.ProjectID, zone, instanceName); err != nil {
+		t.Fatalf("getDeleteProtection got err: %v", err)
+	}
+
+	expectedResult = fmt.Sprintf("Instance %s has DeleteProtection value: %v", instanceName, false)
+	if got := buf.String(); !strings.Contains(got, expectedResult) {
+		t.Errorf("getDeleteProtection got %q, want %q", got, expectedResult)
+	}
+
+	buf.Reset()
+
+	deleteReq := &computepb.DeleteInstanceRequest{
```

Comment:
This should probably move to a Cleanup() function, so that we get more reliable deletion.
Especially since you've used Fatalf() above, which halts test execution and will prevent vm deletion in the event of a test failure.

Reply:
Switched to errorf, so now we will execute deleteInstance in any case (as in another samples)